pkg/virtual/framework/dynamic/doc_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamic_test
+
+import (
+	"context"
+	"errors"
+
+	genericapiserver "k8s.io/apiserver/pkg/server"
+
+	"github.com/kcp-dev/kcp/pkg/apis/apiresource/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic"
+	apidefs "github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/apidefs"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/apiserver"
+)
+
+// Typical Usage of a DynamicVirtualWorkspace
+func Example() {
+
+	var someAPISetRetriever apidefs.APISetRetriever
+	ready := false
+
+	var _ = dynamic.DynamicVirtualWorkspace{
+
+		Name: "SomeDynamicVirtualWorkspace",
+
+		RootPathResolver: func(urlPath string, requestContext context.Context) (accepted bool, prefixToStrip string, completedContext context.Context) {
+			if someAPISetRetriever == nil {
+				// If the APISetRetriever is not initialized, don't accept the request
+				return
+			}
+
+			var apiDomainKey string
+
+			// Resolve the request root path, and extract API domain key from it ...
+
+			if _, exists := someAPISetRetriever.GetAPIs(apiDomainKey); !exists {
+				// If the APISetRetriever doesn't manage this api domain, don't accept the request
```

Comment:
that's an odd place here in the root resolver.

Reply:
Why ? wdym ?